documents/service.go

```diff
@@ -49,3 +53,107 @@ type Service interface {
 	// ReceiveAnchoredDocument receives a new anchored document over the p2p layer, validates and updates the document in DB
 	ReceiveAnchoredDocument(model Model, headers *p2ppb.CentrifugeHeader) error
 }
+
+// service implements Service
+type service struct {
+	config Config
+	repo   Repository
+}
+
+// DefaultService returns the default implementation of the service
+func DefaultService(config config.Configuration, repo Repository) Service {
+	return service{repo: repo, config: config}
+}
+
+func getIds(model Model) ([]byte, []byte, error) {
+	cd, err := model.PackCoreDocument()
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return cd.DocumentIdentifier, cd.NextVersion, nil
+}
+
+func (s service) searchVersion(m Model) (Model, error) {
+
+	id, next, err := getIds(m)
+
+	nm, err := s.getVersion(id, next)
+
+	if err != nil {
+		// here the err is returned as nil because it is expected that the nextVersion
+		// is not available in the db at some stage of the iteration
+		return m, nil
+	}
+
+	if next != nil {
```

Comment:
Next is always set, we dont need this check IMO

Reply:
That's true. Otherwise, it would not be possible to create a nextVersion without modifying the older version. 

Currently, it is just some additional error handling. Something is strange with nextVersion let's  return the actual.